src/components/Calculator.js

```diff
@@ -1,40 +1,76 @@
 import React, { Component } from 'react';
+import calculate from '../logic/calculate';
 
 export default class Calculator extends Component {
+  constructor(props) {
+    super(props);
+    this.handleState = this.handleState.bind(this);
+    this.state = {
+      total: '',
+      next: '',
+      operation: '',
+    };
+  }
+
+  handleState(newState) {
+    if (newState.total != null) {
+      this.setState({ total: newState.total });
+    }
+
+    this.setState({ next: newState.next });
+
+    if (newState.operation != null) {
+      this.setState({ operation: newState.operation });
+    }
+  }
+
+  getInf(state, buttonName) {
+    this.handleState(calculate(state, buttonName));
+  }
+
+  display = (next) => {
+    if (next != null) {
+      return next;
+    }
+    const newnext = '';
+    return newnext;
+  }
+
   render() {
+    const { total, next, operation } = this.state;
     return (
       <div className="calContainer">
-        <input className="calInput" value="0" disabled="disabled" />
+        <input className="calInput" value={`${total} ${operation}${this.display(next)}`} disabled="disabled" />
         <div className="calNumbers">
           <div className="conFirst">
-            <button className="colorW" type="button">AC</button>
-            <button className="colorW" type="button">+/-</button>
-            <button className="colorW" type="button">%</button>
-            <button className="colorS" type="button">+</button>
+            <button className="colorW" type="button" name="AC" onClick={(e) => { this.getInf(this.state, e.target.name); }}>AC</button>
+            <button className="colorW" type="button" name="+/-" onClick={(e) => { this.getInf(this.state, e.target.name); }}>+/-</button>
+            <button className="colorW" type="button" name="%" onClick={(e) => { this.getInf(this.state, e.target.name); }}>%</button>
+            <button className="colorS" type="button" name="÷" onClick={(e) => { this.getInf(this.state, e.target.name); }}>÷</button>
           </div>
           <div className="conSecond">
-            <button className="colorW" type="button">7</button>
-            <button className="colorW" type="button">8</button>
-            <button className="colorW" type="button">9</button>
-            <button className="colorS" type="button">X</button>
+            <button className="colorW" type="button" name="7" onClick={(e) => { this.getInf(this.state, e.target.name); }}>7</button>
+            <button className="colorW" type="button" name="8" onClick={(e) => { this.getInf(this.state, e.target.name); }}>8</button>
+            <button className="colorW" type="button" name="9" onClick={(e) => { this.getInf(this.state, e.target.name); }}>9</button>
+            <button className="colorS" type="button" name="x" onClick={(e) => { this.getInf(this.state, e.target.name); }}>x</button>
           </div>
```

Comment:
- [ ] Check how the given design looks like and try to improve
Design

![design](https://user-images.githubusercontent.com/44745855/129184011-2f7943d4-c63f-4df4-89f2-cf16bfe54ce0.png)

yours

![sc](https://user-images.githubusercontent.com/44745855/129184053-5ba4d4ac-2809-4663-a0b9-8c2c7a68a0d2.png)

- In the calculator, you know that there is always a default `zero` in the screen when you are not doing any operation just add that
- I recognized that after performing an operation let us say `20+20` the response comes with the operator again like `40+` I recommend fixing that It doesn't look nice.. and add a simple space between the operator and the second number